lib/ansible/modules/identity/cyberark/cyberark_user.py

```diff
@@ -0,0 +1,461 @@
+                     "\n*** end_point=%s%s\n ==> %s" % (api_base_url, end_point, exception_text)),
+                headers=headers,
+                status_code=http_exception.code)
+
+    except Exception as unknown_exception:
+
+        module.fail_json(
+            msg=("Unknown error while performing userDelete."
+                 "\n*** end_point=%s%s\n%s" % (api_base_url, end_point, to_text(unknown_exception))),
+            headers=headers,
+            status_code=-1)
+
+
+def userAddToGroup(module):
+
+    # Get username, and groupname from module parameters, and api base url
+    # along with validate_certs from the cyberark_session established
+    username = module.params["username"]
+    group_name = module.params["group_name"]
+    cyberark_session = module.params["cyberark_session"]
+    api_base_url = cyberark_session["api_base_url"]
+    validate_certs = cyberark_session["validate_certs"]
+
+    if module.check_mode:
+        # return if case of check_mode
+        module.exit_json(change=False)
+
+    # Prepare result, end_point, headers and payload
+    result = {}
+    end_point = "/PasswordVault/WebServices/PIMServices.svc//Groups/{0}/Users".format(
+        group_name)
+
+    headers = {'Content-Type': 'application/json'}
+    headers["Authorization"] = cyberark_session["token"]
+    payload = {"UserName": username}
+
+    try:
+
+        # execute REST action
+        response = open_url(
+            api_base_url + end_point,
+            method="POST",
+            headers=headers,
+            data=json.dumps(payload),
+            validate_certs=validate_certs)
+
+        result = {"result": {}}
+
+        return (True, result, response.getcode())
+
+    except (HTTPError, httplib.HTTPException) as http_exception:
+
+        exception_text = to_text(http_exception)
+        if http_exception.code == 409 and "ITATS262E" in exception_text:
+            # User is already member of Group
+            return (False, None, http_exception.code)
+        else:
+            module.fail_json(
+                msg=("Error while performing userAddToGroup."
+                     "Please validate parameters provided."
+                     "\n*** end_point=%s%s\n ==> %s" % (api_base_url, end_point, exception_text)),
+                payload=payload,
+                headers=headers,
+                status_code=http_exception.code)
+
+    except Exception as unknown_exception:
+
+        module.fail_json(
+            msg=("Unknown error while performing userAddToGroup."
+                 "\n*** end_point=%s%s\n%s" % (api_base_url, end_point, to_text(unknown_exception))),
+            payload=payload,
+            headers=headers,
+            status_code=-1)
+
+
+def main():
+
+    fields = {
+        "username": {"required": True, "type": "str"},
+        "state": {"type": "str",
+                  "choices": ["details", "present", "update", "addtogroup", "absent"],
+                  "default": "details"},
```

Comment:
The choices here don't seem to match with the documentation or the code.  The default doesn't match with the code.